Resources/public/js/views/ez-universaldiscoveryview.js

```diff
@@ -279,38 +279,58 @@ YUI.add('ez-universaldiscoveryview', function (Y) {
          * @protected
          */
         _updateMethods: function () {
-            var visibleMethod = this.get('visibleMethod'),
-                startingLocationId = this.get('startingLocationId'),
+            var startingLocationId = this.get('startingLocationId'),
                 startingLocation = this.get('startingLocation'),
                 minDiscoverDepth = this.get('minDiscoverDepth'),
                 virtualRootLocation = this.get('virtualRootLocation');
 
-            /**
-             * Stores a reference to the visible method view
-             *
-             * @property _visibleMethodView
-             * @protected
-             * @type {eZ.UniversalDiscoveryMethodBaseView|Null}
-             */
-            this._visibleMethodView = null;
+            if (this.get('active')) {
+                /**
+                 * Stores a reference to the visible method view
+                 *
+                 * @property _visibleMethodView
+                 * @protected
+                 * @type {eZ.UniversalDiscoveryMethodBaseView|Null}
+                 */
+                this._visibleMethodView = this._getCorrectVisibleMethod();
+
+                Y.Array.each(this.get('methods'), function (method) {
+                    var visible = (method.get('identifier') === this._visibleMethodView.get('identifier'));
+                    method.setAttrs({
+                        'virtualRootLocation': virtualRootLocation,
+                        'multiple': this.get('multiple'),
+                        'loadContent': true,
+                        'minDiscoverDepth': minDiscoverDepth,
+                        'startingLocationId': startingLocationId,
+                        'startingLocation': startingLocation,
+                        'visible': visible,
+                        'isSelectable': Y.bind(this.get('isSelectable'), this),
+                        'active': this.get('active'),
+                    });
+                }, this);
+            }
+        },
+
+        /**
+         * Return the visible method, if a wrong or unknown one is setted, it will return the default one.
+         *
+         * @method _getCorrectVisibleMethod
+         * @protected
+         */
+        _getCorrectVisibleMethod: function () {
```

Comment:
this name does not feel right. Also, on the approach, instead of the double iteration on the available methods, why don't you post check that we found a visible method, and if none was found, you take the first one (or the one indicated in the default value of the `visibleMethod` attribute) ?